### PR TITLE
Closes #604 Migrate to azure blob storage

### LINKED ISSUE #604: Migrate to azure blob storage

### PR DESCRIPTION
**Pull request by AndriiDzendzia**

### CODE REVIEW

**Review by vladzavorotny — APPROVED** [review]

**Review by VasylShtenovych — APPROVED** [review]

File: Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+            var blobContainerClient = blobServiceClient!.GetBlobContainerClient(azureBlobStorageOptions.Value.Container!);
+
+            await using (fileStream)
+            {
+                try
+                {
+                    await using var compressedFile = compressor.CompressFile(fileStream, Quality);
+                    id += ".jpg";
+                    var blobClient = blobContainerClient.GetBlobClient(id);
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs:54]: write some private method for repetitive logic

File: Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            this.azureBlobStorageOptions = azureBlobStorageOptions;
+            this.compressor = compressor;
+            SetCredentials();
+        }
+
+        /// <summary>
+        /// Sets the credentials for the azure-blob-storage service.
+        /// </summary>
+        public void SetCredentials()
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs]: do we really need this method? if so why this is public ?

File: Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            blobServiceClient = new BlobServiceClient(azureBlobStorageOptions.Value.AccessKey!);
```
Comment (serhii-zinkiv) [inline Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs]: do we always sure that AccessKey and Container are not null ? what if they will be null ?

File: Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs (hunk 2 above)
Comment (AndriiDzendzia) [reply Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs]: this method was removed in Add tests for AzureBlobStorageService

File: Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs (hunk 1 above)
Comment (AndriiDzendzia) [reply Car.Backend/Car.Domain/Services/Implementation/AzureBlobStorageService.cs:54]: repetitive logic was removed in #610

**Pull request merged**